t/message-decode-brotlibomb.t

```diff
@@ -0,0 +1,104 @@
+# https://rt.cpan.org/Public/Bug/Display.html?id=52572
+
+use strict;
+use warnings;
+
+use Test::More;
+
+use HTTP::Headers    qw();
+use HTTP::Response   qw();
+
+my $ok = eval {
+    require IO::Compress::Brotli;
+    require IO::Uncompress::Brotli;
+    1;
+};
+if(! $ok) {
+    plan skip_all => "IO::Compress::Brotli needed; $@";
+    exit
+}
+plan tests => 9;
+
+# Create a nasty brotli stream:
+my $size = 16 * 1024 * 1024;
+my $stream = "\0" x $size;
+
+# Compress that stream three times:
+my $compressed = $stream;
+my $bro = IO::Compress::Brotli->create;
+
+for( 1 ) {
+    my $last = $compressed;
+    $compressed = $bro->compress( $compressed );
+    $compressed .= $bro->finish();
+    note sprintf "Encoded size %d bytes after round %d", length $compressed, $_;
+};
+
+my $body = $compressed;
+
+my $headers = HTTP::Headers->new(
+    Content_Type => "application/xml",
+    Content_Encoding => 'br', # only one round needed for Brotli
+);
+my $response = HTTP::Response->new(200, "OK", $headers, $body);
+
+my $len = length $response->decoded_content;
+is($len, 16 * 1024 * 1024, "Self-test: The decoded content length is 16M as expected" );
+
+# Manual decompression check
+my $output = $compressed;
+for( 1 ) {
+    my $unbro = IO::Uncompress::Brotli->create();
+    $output = $unbro->decompress($compressed);
+};
+
+$headers = HTTP::Headers->new(
+    Content_Type => "application/xml",
+    Content_Encoding => 'br' # say my name three times
```

Comment:
It's only there one time

Reply:
Addressed by [58e5573](https://github.com/libwww-perl/HTTP-Message/pull/181/commits/58e5573aac0d74a3c84339ffd56df868afc10e1f)